### ISSUE TITLE
Encrypt and persist the seed in a secure storage

### ISSUE THREAD

**Issue by dpad85** [ACINQ/phoenix-kmm#34]
Current solution on Android is as follows:

#### Encryption

User has 2 options:

1. Use an encryption key stored in the Android Keystore ([see here](https://github.com/ACINQ/phoenix/blob/master/app/src/main/java/fr/acinq/phoenix/utils/crypto/EncryptedSeed.kt#L175))

2. Use an encryption key stored in the Android Keystore **and** unlockable only after Biometric authentication ([see here](https://github.com/ACINQ/phoenix/blob/c161b6b4cea84bf7b40d2b9cb1fa8a633db71884/app/src/main/java/fr/acinq/phoenix/utils/crypto/EncryptedSeed.kt#L187))

Notes: 
- User can freely switch from one option to the other. On Android it means 2 separate encryption keys.
- By default, use option 1 (no biometrics)
- App can be screen locked, which is not related to the seed. It only locks the UI behind Biometrics auth. If using encryption option 2, then this screen lock also unlock the key and decrypt the seed.
- If possible use the device hardware to encrypt the seed (TEE/SE).
- 12-words mnemonics can be retrieved and displayed at runtime 
 - Do not use BIP39 passphrase.

#### Storage

1. seed is [serialized](https://github.com/ACINQ/phoenix/blob/c161b6b4cea84bf7b40d2b9cb1fa8a633db71884/app/src/main/java/fr/acinq/phoenix/utils/crypto/EncryptedSeed.kt#L200) as a `ByteArray`:
a. seed file version (`1` is legacy, `2` is current)
b. type of the key encrypting the memonics
c. initialization vector
d. ciphered mnemonics

2. [`ByteArray` is saved](https://github.com/ACINQ/phoenix/blob/c161b6b4cea84bf7b40d2b9cb1fa8a633db71884/app/src/main/java/fr/acinq/phoenix/utils/crypto/EncryptedSeed.kt#L71) in a file in the private directory of the app. Use a temporary file +  an atomic move.

## Todo iOS

- [ ] Support the 1st encryption option using the iOS keychain. Securing the key behind Biometrics authentication is only a nice to have for now.
- [ ] Support screen lock using device lock
- [ ] Store the seed, either in a file or in a database. Sharing the storage logic between Android and iOS may not be very efficient because it will probably be tightly coupled with the encryption logic which is not shared.
